### ISSUE TITLE
macOS selection is off a character

### ISSUE THREAD

**Issue by ehjmx** [raphamorim/rio#374]
Say I use 
`open -a Safari https://www.google.com` in rio on macOS and decided to select the `https://www.google.com` part with the mouse to copy and paste it in the url bar of Firefox which I already have open the selection is one character off.

If I start in front of the h only `ttps://www.google.com` is select. I need to go close to the end of the i of Safari to select the whole Url `https://www.google.com`.  

I used this as a demo to describe the issue and for this case it's not that important if the selection is one of or not but when I do some work in autopkg or api testing and have to copy long curl urls etc this is annoying. 

macOS 13.6.1
bash: 5.2.21 via macports

**Comment by raphamorim** [comment]
Hi @ehjmx thanks for the issue,

Which rio version and font (family and size) that are you using?

**Comment by ehjmx** [comment]
Hi @raphamorim 

This occured with rio 0.0.29 and is still present in the most recent 0.0.30.

It worked for a while but I don't know which version re-introduced this issue. I remember that it was (almost) gone and the best in a while with 0.0.28 but present in versions prior to 0.0.28.

I mainly use Fira Code. Recently I used Commit Mono as well because for this font I get the weight which I deemed helpful in regards of issue #372.

**Comment by raphamorim** [comment]
I will close this issue since might prob was fixed on 0.1.x releases like https://github.com/raphamorim/rio/issues/535. The 0.1.x have a completely new renderer so is hard to follow up on this issue context. In case it still happening on 0.1.x, will be better create a new issue.

**Comment by ehjmx** [comment]
With 0.1.6 everything was working as expected, no offset when selecting a command. Now with 0.1.7 at least for me, the offset is back. I'll see if I can make screenshots or a little screenrecording after my working hours.
macOS 14.6.1
bash 5.2.32
Font: Fira Code 6.2
